pkg/ingester/ingester_v2.go

```diff
@@ -767,7 +767,10 @@ func (i *Ingester) v2Push(ctx context.Context, req *client.WriteRequest) (*clien
 	}
 	i.TSDBState.appenderCommitDuration.Observe(time.Since(startCommit).Seconds())
 
-	db.setLastUpdate(time.Now())
+	// If only invalid samples are pushed, don't change "last update", as TSDB was not modified.
+	if succeededSamplesCount > 0 {
+		db.setLastUpdate(time.Now())
```

Comment:
I don't think we need to avoid `time.Now()` here. Furthermore there are cases where commit takes quite a while. We only have bucket up to 10 seconds, but metrics show that we hit it regularly on some ingesters (!), although in practice that should still be good enough.